### PR TITLE
Fixes tests for iOS 17

### PR DESCRIPTION
**Pull request by guykogus**
🚀
Some tests fail in iOS 17 and need to be updated. E.g. `URL(string:)` has changed the way that it parses strings.
> For apps linked on or after iOS 17 and aligned OS versions, [NSURL](https://developer.apple.com/documentation/foundation/nsurl) parsing has updated from the obsolete RFC 1738/1808 parsing to the same [RFC 3986](https://www.ietf.org/rfc/rfc3986.txt) parsing as [NSURLComponents](https://developer.apple.com/documentation/foundation/nsurlcomponents). This unifies the parsing behaviors of the NSURL and NSURLComponents APIs. Now, NSURL automatically percent- and IDNA-encodes invalid characters to help create a valid URL.

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.0.
- [x] New extensions support iOS 10.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

**Comment by guykogus** [comment]
@LucianoPAlmeida I fixed most of the tests, but some are not working. Wanna have a look and play with it? It looks like in iOS 17 the way images are created using the renderer have extra data somewhere inside that creates mismatches.

File: Sources/SwifterSwift/UIKit/UIImageExtensions.swift [hunk 1]
```diff
@@ -102,11 +102,23 @@ public extension UIImage {
...
-        UIGraphicsBeginImageContextWithOptions(CGSize(width: newWidth, height: toHeight), opaque, self.scale)
-        draw(in: CGRect(x: 0, y: 0, width: newWidth, height: toHeight))
-        let newImage = UIGraphicsGetImageFromCurrentImageContext()
-        UIGraphicsEndImageContext()
-        return newImage
+        let size = CGSize(width: newWidth, height: toHeight)
+        let rect = CGRect(origin: .zero, size: size)
+
+        #if !os(watchOS)
+        if #available(tvOS 10.0, *) {
```
Comment (LucianoPAlmeida) [inline Sources/SwifterSwift/UIKit/UIImageExtensions.swift:109]: We have iOS 10 as minimum target deployment for the project, which personally think is too old, but I think we can just drop the old code and remove the availability checks across. Maybe even move tvOS to 10 as well. But that is another discussion

**Comment by guykogus** [comment]
@LucianoPAlmeida looks like I finally fixed all the tests and compilations and swiftlint and whatnot. Will you have a check?

And what about deployment versions? Should we use the minimum recommended by Xcode 15? E.g. iOS/tvOS 12? We can do that in a separate PR.

**Comment by LucianoPAlmeida** [comment]
> Should we use the minimum recommended by Xcode 15? E.g. iOS/tvOS 12? We can do that in a separate PR.

I think yes, we can definitely experiment with that in another PR :)

**Pull request merged**